Mathlib/Analysis/NormedSpace/MStructure.lean

```diff
@@ -311,3 +312,58 @@ instance Subtype.BooleanAlgebra [FaithfulSMul M X] :
     sdiff_eq := fun P Q => Subtype.ext <| by rw [coe_sdiff, ← coe_compl, coe_inf] }
 
 end IsLprojection
+
+section WithL1
+
+open ENNReal
+
+variable (p : ℝ≥0∞) (α β : Type*)
+
+variable {p α β}
+variable [NormedAddCommGroup α] [NormedAddCommGroup β]
+
+namespace WithLp
+
+/-- Projection on `WithLp p (α × β)` with range `α` and kernel `β` -/
+def idemFst : AddMonoid.End (WithLp p (α × β)) := (AddMonoidHom.inl α β).comp (AddMonoidHom.fst α β)
+
+/-- Projection on `WithLp p (α × β)` with range `β` and kernel `α` -/
+def idemSnd : AddMonoid.End (WithLp p (α × β)) := (AddMonoidHom.inr α β).comp (AddMonoidHom.snd α β)
+
+lemma idemFst_apply (x : WithLp p (α × β)) : idemFst x = (WithLp.equiv _ _).symm (x.1, 0) := rfl
+
+lemma idemSnd_apply (x : WithLp p (α × β)) : idemSnd x = (WithLp.equiv _ _).symm (0, x.2) := rfl
+
+lemma idemFst_compl : (1 : AddMonoid.End (WithLp p (α × β))) - idemFst = idemSnd := AddMonoidHom.ext
+  fun x => by
+    rw [AddMonoidHom.sub_apply, idemFst_apply, idemSnd_apply, AddMonoid.End.coe_one, id_eq,
+      sub_eq_iff_eq_add, add_comm, ← WithLp.equiv_symm_add, Prod.mk_add_mk, zero_add, add_zero]; rfl
+
+theorem prod_norm_eq_sup_idemFst (x : WithLp ∞ (α × β)) :
+    ‖x‖ = ‖idemFst x‖ ⊔ ‖(1 - idemFst) x‖ := by
```

Comment:
I'd be inclined to write this as
```suggestion
    ‖x‖ = ‖idemFst x‖ ⊔ ‖idemSnd x‖ := by
```
which is shorter and more symmetric. Same below.

Reply:
`P` and `1-P` tends to be the L-projection way of doing things, but I can see why `Fst` and `Snd` makes sense in this case.

Reply:
I claim the L-projection way of doing things is now `IsLprojection` :)